### PR TITLE
Require GROUP, POM_ARTIFACT_ID & VERSION_NAME to be set and fail on Gradle < 4.10.1

### PR DESCRIPTION
**Pull request by martinbonnin**
Closes https://github.com/vanniktech/gradle-maven-publish-plugin/issues/61

I made all the gradle properties mandatory except `POM_DESCRIPTION`. Maybe some others are optional ?

### CODE REVIEW

File: src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt [hunk 1]
```diff
@@ -36,6 +38,17 @@ internal abstract class BaseMavenPublishPlugin : Plugin<Project> {
     }
   }
 
+  private fun checkGradleVersion(p: Project) {
```
Comment (vanniktech) [inline src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt]: You can use `VersionNumber.parse` and then use `>=` to compare.

File: src/main/kotlin/com/vanniktech/maven/publish/MavenPublishPom.kt [hunk 2]
```diff
@@ -3,45 +3,55 @@ package com.vanniktech.maven.publish
...
+        findMandatoryProperty(project, "POM_LICENCE_URL"),
+        findMandatoryProperty(project, "POM_LICENCE_DIST"),
+        findMandatoryProperty(project, "POM_DEVELOPER_ID"),
+        findMandatoryProperty(project, "POM_DEVELOPER_NAME")
     )
+
+    private fun findMandatoryProperty(project: Project, propertyName: String): String {
+      val value = project.findProperty(propertyName) as String?
+      if (value == null) {
+        throw IllegalArgumentException("please define $propertyName in your gradle.properties")
```
Comment (vanniktech) [inline src/main/kotlin/com/vanniktech/maven/publish/MavenPublishPom.kt]:
```suggestion
        throw IllegalArgumentException("Please define \"$propertyName\" in your gradle.properties file")
```

File: src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt (hunk 1 above)
Comment (martinbonnin) [reply src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt]: Ah, thanks, I was wondering if such a thing existed!

**Comment by martinbonnin** [comment]
Remaining tests fail on `SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable`. They also fail on master. did something change on travis recently ?

**Comment by martinbonnin** [comment]
Right, they fail on my machine due to the android SDK but the latest error is something else indeed.

**Comment by gabrielittner** [comment]
Will take a look tomorrow.

**Comment by martinbonnin** [comment]
Travis should be alright now. It was a ktlint issue, not and Android Sdk one indeed. Sorry for the noise.

File: src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt [hunk 3]
```diff
@@ -3,12 +3,18 @@ package com.vanniktech.maven.publish
...
     val extension = p.extensions.create("mavenPublish", MavenPublishPluginExtension::class.java, p)
 
+    val gradleVersion = VersionNumber.parse(p.gradle.gradleVersion)
+    if (gradleVersion < VersionNumber(MINIMUM_GRADLE_MAJOR, MINIMUM_GRADLE_MINOR, MINIMUM_GRADLE_MICRO, null)) {
+      throw IllegalArgumentException("You need gradle version 4.10.1 or higher")
+    }
+
     val pom = MavenPublishPom.fromProject(p)
     p.group = requireNotNull(pom.groupId) { "groupId is required to be set" }
     p.version = requireNotNull(pom.version) { "version is required to be set" }
```
Comment (gabrielittner) [inline src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt]: Can you remove these two lines? They shouldn't happen anymore.

Comment (martinbonnin) [reply src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt]: indeed, thanks

**Pull request merged**